src/game/shell.c

```diff
@@ -194,20 +271,40 @@ bool Shell_MakeScreenshot()
         break;
     }
 
-    char path[20];
-    for (int i = 0; i < 10000; i++) {
-        sprintf(path, "screenshot%04d.%s", i, ext);
+    // Screenshot folder
+    char ss_folder[SS_FOLDER_NAME_SIZE];
+    sprintf(ss_folder, "screenshots");
+    File_CreateDirectory(ss_folder);
+
+    // Screenshot name
+    char ss_name[TIMESTAMP_SIZE + LEVEL_TITLE_SIZE];
+    Shell_GetScreenshotName(ss_name);
+
+    // Screenshot folder/name path
+    char path
+        [SS_FOLDER_NAME_SIZE + TIMESTAMP_SIZE + LEVEL_TITLE_SIZE
+         + EXTENSION_SIZE];
+    sprintf(path, "%s/%s.%s", ss_folder, ss_name, ext);
+    char *full_path = NULL;
+    File_GetFullPath(path, &full_path);
+    if (!File_Exists(full_path)) {
+        bool result = Output_MakeScreenshot(full_path);
+        Memory_FreePointer(full_path);
```

Comment:
This should crash, we're supposed to pass variables by reference to `Memory_FreePointer` (almost always)